xgboost-sys/build.rs

```diff
@@ -22,10 +22,29 @@ fn main() {
     }
 
     // CMake
+    #[cfg(feature = "cuda")]
     let dst = Config::new(&xgb_root)
         .uses_cxx11()
         .define("BUILD_STATIC_LIB", "ON")
-        .build();
+        .define("USE_CUDA", "ON")
+        .define("BUILD_WITH_CUDA", "ON")
+        .define("BUILD_WITH_CUDA_CUB", "ON");
+
+    #[cfg(not(feature = "cuda"))]
+    let mut dst = Config::new(&xgb_root);
+
+    let dst = dst.uses_cxx11()
+        .define("BUILD_STATIC_LIB", "ON");
+
+    #[cfg(target_os = "macos")]
+    let dst =
+        dst
+            .define("CMAKE_C_COMPILER", "/opt/homebrew/opt/llvm/bin/clang")
+            .define("CMAKE_CXX_COMPILER", "/opt/homebrew/opt/llvm/bin/clang++")
+            .define("OPENMP_LIBRARIES", "/opt/homebrew/opt/llvm/lib")
+            .define("OPENMP_INCLUDES", "/opt/homebrew/opt/llvm/include");
```

Comment:
It would be good to leave this as configurable (to allow for non-homebrew LLVM installs), and to allow for homebrew installs to different locations.

Homebrew defaults to `/usr/local` for intel macs, and `/opt/homebrew` for silicon: https://docs.brew.sh/FAQ#why-should-i-install-homebrew-in-the-default-location, so bound to be some variation here.

I had to comment out the above to compile, since I'm on an intel mac (hoping for an m1 someday...).

Reply:
I added a check for the m1 directory existence before adding these extras in https://github.com/davechallis/rust-xgboost/pull/21/commits/1da1c032e37ba4ecdc2808fe16dfd663034bdb71

Let me know if this is still causing an issue for intel. I don't have a laptop I can test this one on.